packages/@lwc/features/src/babel-plugin/index.js

```diff
@@ -38,6 +38,11 @@ module.exports = function({ types: t }) {
                     const didImportRuntimeFlags = specifiers.some(specifier => {
                         return specifier.local && specifier.local.name === RUNTIME_FLAGS_IDENTIFIER;
                     });
+                    if (didImportRuntimeFlags && !this.opts.prod) {
```

Comment:
Why is it only scoped to non-prod environments?

Reply:
This is because we apply this transform on code that has already been modified for non-prod mode. In non-prod mode, we change everything to runtime flags for testing purposes.